### PR TITLE
Update testcafe

### PR DESCRIPTION
**Pull request by Argn0**
testcafe runs for 2m 48s on travis with these changes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,8 +9,8 @@
     "build": "react-scripts build",
     "build:install": "npm install && npm run lintnofix && npm run build",
     "snapshot": "react-snap",
-    "test": "react-scripts test --env=jsdom",
-    "testcafe": "testcafe -c 1 chrome testcafe/ --app \"react-scripts start\" --app-init-delay 30000",
+    "test": "testcafe -c 4 chrome:headless testcafe/ --app \"serve -s build\"",
```
Comment (howardchung) [inline package.json]: headless doesn't work on travis?

File: src/actions/action.js [hunk 2]
```diff
@@ -17,14 +17,17 @@ export default function action(type, host, path, params = {}, transform) {
...
+          err.fetchError = true;
           dispatch(getError(response.status));
           if (response.status >= 400 && response.status < 500) {
-            const err = new Error('client error');
-            err.clientError = true;
+            err.message = 'fetch failed - client error';
             throw err;
           } else {
-            throw new Error('fetch failed');
+            setTimeout(() => fetchDataWithRetry(delay + 3000), delay);
```
Comment (howardchung) [inline src/actions/action.js]: if we're retrying why are we also throwing the error?

File: package.json (hunk 1 above)
Comment (Argn0) [reply package.json]: headless mode had slightly worse or same performance on travis when I tested it.

File: src/actions/action.js (hunk 2 above)
Comment (Argn0) [reply src/actions/action.js]: I throw here so the promise chain jumps to the catch handler.

File: package.json (hunk 1 above)
Comment (howardchung) [reply package.json]: if it's approx the same I'd rather keep one version of the test to avoid fragmentation

File: src/actions/action.js (hunk 2 above)
Comment (howardchung) [reply src/actions/action.js]: I think we should only throw if we exceed some retry threshold.

Comment (Argn0) [reply src/actions/action.js]: But if I don't throw here, then the next then()  ``` .then(transform || (json => json))```  will throw anyway

Comment (howardchung) [reply src/actions/action.js]: hmm, ok.  In that case I think we should move the retry back to the catch block.

**Pull request merged**